modules/openapi-generator/src/main/resources/cpp-pistache-server/model-source.mustache

```diff
@@ -18,13 +18,60 @@ namespace {{this}} {
     {{/required}}{{/vars}}
 }
 
-{{classname}}::~{{classname}}()
+void {{classname}}::validate() const
 {
+    std::stringstream msg;
+    if (!validate(msg))
+    {
+        throw {{helpersNamespace}}::ValidationException(msg.str());
+    }
 }
 
-void {{classname}}::validate()
+bool {{classname}}::validate(std::stringstream& msg) const
 {
-    // TODO: implement validation
+    return validate(msg, "");
+}
+
+bool {{classname}}::validate(std::stringstream& msg, const std::string& pathPrefix) const
+{
+    bool success = true;
+    const std::string _pathPrefix = pathPrefix.empty() ? "{{classname}}" : pathPrefix;
+
+    {{#isEnum}}{{! Special case for enum types }}
+    if (m_value == {{classname}}::e{{classname}}::INVALID_VALUE_OPENAPI_GENERATED)
+    {
+        success = false;
+        msg << _pathPrefix << ": has no value;";
+    }
+    {{/isEnum}}
+    {{^isEnum}}
+    {{#vars}}
+    {{^required}}if ({{nameInCamelCase}}IsSet()){{/required}}
+    {{#required}}/* {{name}} */ {{/required}}{
+        const {{{dataType}}}& value = m_{{name}};
+        const std::string currentValuePath = _pathPrefix + ".{{nameInCamelCase}}";
+        {{> model-validation-body }}
```

Comment:
I wonder if it's possible to detect here if any content has been generated by model-validation-body. Only thing I don't like here is that it generates a lot of if branches for things, that can't be validated. 

@wing328 maybe you know a way to do it? I tried experimenting with hasValidation property, but it's always false for me.

Reply:
One way is to do it in the generator's (java) model post-processing function to add something like `x-cpp-has-validation` to indicate whether there's any validation

Reply:
There is a field called `has-validation`, I'm not sure if we can use it in this particular context, I will experiment with it.

Reply:
Alright I tested the `hasValidation` a little.
In principle it works, I can check the value and only generate the `const {{{dataType}}}& value....` if it is true.
However, this will break for arrays. Given the following example:
```yaml
myOtherArray:
  type: array
  items:
    type: integer
    format: int64
    minimum: 10
    maximum: 100
```
`hasValidation` remains false for `myOtherArray`, since the array itself does not have validation, but the items inside of it do.
I don't think there's a way to recursively check if the items of an array (which of course can be arrays too) have some sort of validation, without modifying the java code for the openapi-generator. I would try to do that, but it's a fairly large codebase and I don't know my way around it overall.

But I can use this for all other checks, so the unnecessary checks are only generated for array types.

Reply:
We can improve this in the future, You can list the limitations in the Description